bug/disk_logger

BarAgam opened the pull request:
Ended up as few different bug fixes:
1. DiskLogger - now retries log write to file (3 times) instead of throwing an exception. It will use the native logger to log failures instead of throwing and crashing the app.
2. LocationFetcher - Checks provider is supported by the device and enabled. It'll throw an exception if it isn't. LauncherActivity now tries to add the different providers.
3. UserLocationViewModel - will update user's location geometry and not just its symbology. Users' pins will move over the map now.

Other things:
1. Implemented a LoggerFactory create method without arguments that uses a tag extract by itself. (currently using the calling class simple name as the tag)
2.  Moved constants out of the DiskLogger class into Constants classes.
3. Moved a few methods to meet organization conventions

And merged BroadcastReceiver unregistration fix

gRaytan committed:
unregister mGpsStatusAlertBroadcastReceiver in MainActivity.onPause

barmako committed:
Ended up as few different bug fixes:
 1. DiskLogger - now retries log write to file (3 times) instead of throwing an exception. It will use the native logger to log failures instead of throwing and crashing the app.
 2. LocationFetcher - Checks provider is supported by the device and enabled. It'll throw an exception if it isn't. LauncherActivity now tries to add the different providers.
 3. UserLocationViewModel - will update user's location geometry and not just its symbology. Users' pins will move over the map now.

 Other things:
 1. Implemented a LoggerFactory create method without arguments that uses a tag extract by itself. (currently using the calling class simple name as the tag)
 2.  Moved constants out of the DiskLogger class into Constants classes.
 3. Moved a few methods to meet organization conventions

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/57" target="_blank"><img src="https://app.review.ninja/52010471/pull/57/badge" alt="ReviewNinja"/>

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java:
```diff
@@ -141,10 +140,14 @@ public void addProvider(@ProviderType String locationProvider) {
         }
 
+        if (!isProviderExistsAndEnabled(locationProvider)) {
+            throw new RuntimeException(
```
Should we throw exception or just notify the user / log?

teamagam3 replied:
!fixed

barmako committed:
Fixed tests.
Added one more test to ensure providers that aren't enabled will cause location fetcher to throw an exception
